### PR TITLE
feat: add default timeout for waiting mined transactions

### PR DESCRIPTION
**Pull request by graemecode**

### CODE REVIEW

File: src/apis/blockchain_api.ts [hunk 1]
```diff
@@ -71,6 +82,10 @@ export class BlockchainAPI {
         const intervalManager = this.intervalManager;
         const web3Utils = this.web3Utils;
 
+        if (!_.isNumber(timeoutMs)) {
```
Comment (saturnial) [inline src/apis/blockchain_api.ts]: why not specify this default value in the signature and remove the optional

Comment (graemecode) [reply src/apis/blockchain_api.ts]: Oh I forgot you could do that lol

Comment (saturnial) [reply src/apis/blockchain_api.ts]: that's what friends are for!

File: src/apis/blockchain_api.ts [hunk 2]
```diff
@@ -17,6 +19,14 @@ export const BlockchainAPIErrors = {
                          with hash ${txHash}.`,
 };
 
+/**
+ * The following default timeout is provided to the IntervalManager when awaiting mined
+ * transactions. The value is represented in milliseconds.
+ *
+ * @type {number}
+ */
+export const DEFAULT_TIMEOUT_FOR_TX_MINED = 60000;
```
Comment (saturnial) [inline src/apis/blockchain_api.ts]: Can we add a bit more color as to why this value was chosen? What is it's significance

Comment (graemecode) [reply src/apis/blockchain_api.ts]: Hmmm, it may be arbitrary - it's what we currently use in Plex; that's where i got it from.

**Pull request merged**